src/Microsoft.AspNet.OData.Shared/DeltaDeletedEntityObjectOfT.cs

```diff
@@ -0,0 +1,121 @@
+            : this(structuralType, updatableProperties: null, dynamicDictionaryPropertyInfo, instanceAnnotationsPropertyInfo)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="DeltaDeletedEntityObject{TStructuralType}"/>.
+        /// </summary>
+        /// <param name="structuralType">The derived entity type or complex type for which the changes will be tracked.
+        /// <paramref name="structuralType"/> should be assignable to instances of <typeparamref name="TStructuralType"/>.
+        /// </param>
+        /// <param name="updatableProperties">Properties that can be updated.</param>
+        /// <param name="dynamicDictionaryPropertyInfo">The property info that is used as dictionary of dynamic
+        /// properties. <c>null</c> means this entity type is not open.</param>
+        /// <param name="instanceAnnotationsPropertyInfo">The property info that is used as container for instance annotations</param>
+        public DeltaDeletedEntityObject(Type structuralType, IEnumerable<string> updatableProperties, PropertyInfo dynamicDictionaryPropertyInfo, PropertyInfo instanceAnnotationsPropertyInfo)
+            : this(structuralType, updatableProperties, dynamicDictionaryPropertyInfo, false, instanceAnnotationsPropertyInfo)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="DeltaDeletedEntityObject{TStructuralType}"/>.
+        /// </summary>
+        /// <param name="structuralType">The derived entity type or complex type for which the changes will be tracked.
+        /// <paramref name="structuralType"/> should be assignable to instances of <typeparamref name="TStructuralType"/>.
+        /// </param>
+        /// <param name="updatableProperties">Properties that can be updated.</param>
+        /// <param name="dynamicDictionaryPropertyInfo">The property info that is used as dictionary of dynamic
+        /// properties. <c>null</c> means this entity type is not open.</param>
+        /// <param name="isComplexType">To determine if the entity is a complex type</param>
+        /// <param name="instanceAnnotationsPropertyInfo">The property info that is used as container for instance annotations</param>
+        public DeltaDeletedEntityObject(Type structuralType, IEnumerable<string> updatableProperties, PropertyInfo dynamicDictionaryPropertyInfo, bool isComplexType, PropertyInfo instanceAnnotationsPropertyInfo)
+            : base(structuralType, updatableProperties, dynamicDictionaryPropertyInfo, isComplexType, instanceAnnotationsPropertyInfo)
+        {
+            DeltaKind = EdmDeltaEntityKind.DeletedEntry;
+        }
+
+        /// <inheritdoc />
+        public Uri Id { get; set; }
+
+        /// <inheritdoc />
+        public DeltaDeletedEntryReason? Reason { get; set; }
+
+        /// <inheritdoc />
+        public IEdmNavigationSource NavigationSource { get; set; }
```

Comment:
why do we include "Edm" in the C# type?

Reply:
This property is used when serializing a deletedresource. But we just do a check for whether the navigation source is null or not. If not null we create an `odataresourceserializationinfo` for it. and pass in the navigation name..

Reply:
Yes I believe we can use a string here instead of  an `IEdmNavigationSource `